ark/internal/genai/tools.go

```
@@ -381,6 +381,60 @@ func CreateToolFromCRD(toolCRD *arkv1alpha1.Tool) ToolDefinition {
 	return ToolDefinition{Name: toolCRD.Name, Description: description, Parameters: parameters}
 }
 
+func CreatePartialToolDefinition(tooldefinition ToolDefinition, partial *arkv1alpha1.ToolPartial) ToolDefinition {
+	if partial == nil {
+		return tooldefinition
+	}
+
+	newName := tooldefinition.Name
+	newDesc := tooldefinition.Description
+
+	// Deep copy parameters map
+	newParams := map[string]any{}
+	maps.Copy(newParams, tooldefinition.Parameters)
+
+	if partial.Name != "" {
+		newName = partial.Name
+	}
+
+	if partial.Description != "" {
+		newDesc = partial.Description
+	}
+
+	// Remove partial parameters from schema if present in context
+	if props, ok := newParams["properties"].(map[string]any); ok {
```

if we are ok for a silent failure if `properties` doesn't exist pls note this and why, or return an error (otherwise reading it, it can look like a miss and error hiding)

---

Is properties always present? Even if a tool does not have params?